components/scim/org.wso2.carbon.identity.scim.common/src/main/java/org/wso2/carbon/identity/scim/common/listener/SCIMUserOperationListener.java

```diff
@@ -396,8 +396,9 @@ public boolean doPostUpdateRoleName(String roleName, String newRoleName,
             if (domainName == null) {
                 domainName = UserCoreConstants.PRIMARY_DEFAULT_DOMAIN_NAME;
             }
-            String roleNameWithDomain = domainName + CarbonConstants.DOMAIN_SEPARATOR + roleName;
-            String newRoleNameWithDomain = domainName + CarbonConstants.DOMAIN_SEPARATOR + newRoleName;
+            String roleNameWithDomain =
+                    domainName.toUpperCase() + CarbonConstants.DOMAIN_SEPARATOR + roleName;
+            String newRoleNameWithDomain = domainName.toUpperCase() + CarbonConstants.DOMAIN_SEPARATOR + newRoleName;
```

Comment:
User UserCoreUtil.addDomainToName() method instead of uppercasing the domain manually